tests/test_integration.py

```diff
@@ -100,22 +100,21 @@
 
 
 @pytest.mark.asyncio
-@pytest.mark.xfail  # FIXME: docker websocket seems not working as expected
 async def test_stdio_stdin(docker, testing_images, shell_container):
     ws = await shell_container.websocket(stdin=True, stdout=True, stream=True)
-    await ws.send_str('echo "hello world"\n')
-    with aiohttp.Timeout(2):
-        # TODO: fix timeout
-        resp = await ws.receive()
-    print(resp)
-    assert resp.data == b"hello world\n"
+    await ws.send_str('echo hello world\n')
+
+    resp = await ws.receive()
+    assert resp.data == "echo hello world\r\n"
     await ws.close()
-    # TODO: ensure container stopped on its own because stdin was closed
 
     # Cross-check container logs.
+    await asyncio.sleep(1)
```

Comment:
good point, you can do something like 
https://github.com/aio-libs/aiodocker/blob/959c2513447c8c83ae169dff70ef898558d0453b/aiodocker/docker.py#L116-L120